### PR TITLE
Implemented: code to make the data dynamic on the completed page(#1xx6t61)

### LINKED ISSUE #93: Make orders information dynamic on completed page
## What are the acceptance criteria?

- [ ] ...

## Can you complete this feature request by yourself?

- [x] YES
- [ ] NO

### PR DESCRIPTION
**Pull request by ymaheshwari1**
### Related Issues

Closes #93 

### Screenshots of Visual Changes before/after (If There Are Any)

**IMPORTANT NOTICE** - Remember to add changelog entry

### Contribution and Currently Important Rules Acceptance

- [x] I read and followed [contribution rules](https://github.com/hotwax/ionic-bopis#contribution-guideline)

### CODE REVIEW

File: src/views/Completed.vue [hunk 1]
```diff
@@ -3,192 +3,212 @@
...
-            <ion-item lines="none">
-              <ion-thumbnail slot="start">
-                <img src="https://dev-resources.hotwax.io/resources/uploads/images/product/m/j/mj08-blue_main.jpg" />
-              </ion-thumbnail>
+            <div class="order-metadata">
               <ion-label>
-                <p class="overline">WJ06-XL-PURPLE</p>
-                Juno Jacket
-                <p>Blue XL</p>
+                {{ orders.doclist.docs[0].shipmentMethodTypeDesc }}
```
Comment (adityasharma7) [inline src/views/Completed.vue:69]: We will not be accessing data with doclist properties, we should transform and use order & items. As discussed to be taken care in upcoming PR

File: src/views/Completed.vue [hunk 2]
```diff
@@ -3,192 +3,212 @@
...
               </ion-label>
-            </ion-item>
+            </div>
           </div>
 
-          <div class="product-metadata mobile-only">
-            <ion-note>49 {{ $t("pieces in stock") }}</ion-note>
-          </div>
-        </div>
+          <div v-for="order in orders.doclist.docs" :key="order" class="order-item">
```
Comment (adityasharma7) [inline src/views/Completed.vue:77]: Here orders is having order and order is having item. Taken care in next PR

File: src/views/Completed.vue [hunk 3]
```diff
@@ -209,6 +229,148 @@ export default defineComponent({
...
+            "facet": {
+              "groups": "unique(orderId)",
+              "itemCount": "sum(itemQuantity)"
+            }
+          }
+        }
+      })
+
+      try {
+        const resp = await UtilService.fetchShipmentMethods(payload)
```
Comment (adityasharma7) [inline src/views/Completed.vue:263]: Check if we need reset of values

File: src/views/Completed.vue [hunk 4]
```diff
@@ -209,6 +229,148 @@ export default defineComponent({
...
+
+        if(resp.status == 200 && !hasError(resp)) {
+          this.shipmentMethods = resp.data.facets.shipmentMethodFacet.buckets
+        }
+      } catch(err) {
+        console.error(err)
+      }
+    },
+
+    async fetchManifestedCarrierPartyIds() {
```
Comment (adityasharma7) [inline src/views/Completed.vue:273]: We could all the carrier partyId in single API call

**Review by adityasharma7 — CHANGES_REQUESTED** [review]

File: src/views/Completed.vue (hunk 2 above)
Comment (ymaheshwari1) [reply src/views/Completed.vue:77]: https://github.com/hotwax/fulfillment-pwa/pull/98/files#diff-4fd9ea4fa1ac3bf01bf1a44d1a3e45717e8108f5fb8549ae83a281680486c966R82

File: src/views/Completed.vue (hunk 3 above)
Comment (ymaheshwari1) [reply src/views/Completed.vue:263]: The component gets unmounted which automatically reset the values, thus we does not need to handle it.

File: src/views/Completed.vue [hunk 5]
```diff
@@ -209,6 +217,102 @@ export default defineComponent({
...
+              "groups": "unique(picklistBinId)"
+            }
+          }
+        }
+      })
+
+      try {
+        const resp = await UtilService.fetchCarrierPartyIds(payload)
+
+        if(resp.status == 200 && !hasError(resp) && resp.data.facets.count >= 0) {
```
Comment (adityasharma7) [inline src/views/Completed.vue:291]:
```suggestion
        if(resp.status == 200 && !hasError(resp)) {
```

**Review by adityasharma7 — APPROVED** [review]

**Pull request merged**